drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/DataBuilder.java

```diff
@@ -62,10 +62,23 @@ public DataBuilder(final Sheet sheet,
     public void build() {
 
         for (final List<DTCellValue52> row : dtable.getData()) {
-            int sourceColumnIndex = 0;
-            int targetColumnIndex = -1;
-            final Row xlsRow = sheet.createRow(rowCount);
-            final List<String> addedInserts = new ArrayList<>();
+            new DataRowBuilder(row).build();
+        }
+    }
+
+    class DataRowBuilder {
+
+        private final Row xlsRow = sheet.createRow(rowCount);
+        private final List<String> addedInserts = new ArrayList<>();
```

Comment:
As we use `addedInserts` just for checking a presence of elements I think we should use `private final Set<String> addedInserts = new HashSet<>();`

Reply:
Done